src/mongo/jobs/DiscoverOutdatedComposites.class.php

```diff
@@ -0,0 +1,296 @@
+     *
+     * @param CompositeMetadata $metadatum - the composite collection to search
+     * @return CompositeRegenTask|null - a CompositeRegenTask, if outdated documents are found; null otherwise
+    **/
+    public function getRegenTaskForMetadata($metadatum, $cursorLimit) {
+        $filterOutdated = $metadatum->getOutdatedQueryComponent();
+        if(!isset($filterOutdated)) {
+            return null;
+        } else {
+            $compositeCollection = $metadatum->compositeCollection;
+            $specification = $metadatum->specification;
+            $cbdCollection = $metadatum->cbdCollection;
+            $compositeRegenFunction = $metadatum->compositeRegenFunction;
+
+            $cbdIdFields = array(_ID_KEY.'.'._ID_RESOURCE => 1, _ID_KEY.'.'._ID_CONTEXT => 1);
+
+            // find one cursor page at a time for this composite
+            $outdatedComposites = 
+                $compositeCollection
+                    ->find($filterOutdated, $cbdIdFields)
+                    ->limit($cursorLimit);
+
+            if(!$outdatedComposites->hasNext()) {
+                // no outdated composite documents found, so no work to be done
+                return null;
+            } else {
+                // find IDs root CBDs required to regenerate the composite
+                $outdatedCbdIds = 
+                    array_map(array($this, 'composite2cbdId'), iterator_to_array($outdatedComposites, false));
+
+                // fetch the CBDs themselves
+                $filterCbdsById = array('$or' => $outdatedCbdIds);
+                $cbdDocs = $cbdCollection->find($filterCbdsById);
+
+                if(!$cbdDocs->hasNext()) {
+                    // no CBDs found, so nothing to regenerate
+                    return null;
+                } else {
+                    // return a task to regenerate outdated composite documents
+                    return new CompositeRegenTask(
+                        $compositeRegenFunction, 
+                        $specification, 
+                        $compositeCollection, 
+                        iterator_to_array($cbdDocs, false)
+                    );
+                }
+            }
+        }
+    }
+
+    // TODO: this should really live in IComposite implementations, but cannot yet.
+    public function makeViewRegenFunc($storeName) {
+        // TODO: remove $views from here. We need it because that class provides a method for
+        // regenerating indvidual views.  However, that should probably be abstracted somewhere,
+        // into a method capable of regenerating individual composites, by type.
+
+        // TODO: these fakes are only here to satisfy Views, which probably does not
+        // need either of these as constructor arguments (most methods in Views are agnostic)
+        $fakeCollection = \Tripod\Mongo\Config::getInstance()->getCollectionForCBD($storeName, 'CBD_testing');
+        $fakeDefaultContext = 'http://talisaspire.com/';
```

Comment:
These are hard-coded to demonstrate issues with constructor calls, and must not make it into production.

* [ ] Remove hard-coded values for Views constructor